txpipe/theory.py

```diff
@@ -103,81 +110,85 @@ def replace_with_theory_real(self, s, cosmo):
+                    )
+                    xip = pyccl.correlation(cosmo, ell, cl, theta / 60, corr_type="L+")
+                    xim = pyccl.correlation(cosmo, ell, cl, theta / 60, corr_type="L-")
+
+                    # replace data values in the sacc object for the theory ones
+                    ind_xip = s.indices(
+                        "galaxy_shear_xi_plus", (f"source_{i}", f"source_{j}")
+                    )
+                    ind_xim = s.indices(
+                        "galaxy_shear_xi_minus", (f"source_{i}", f"source_{j}")
+                    )
+                    for p, q in enumerate(ind_xip):
+                        s.data[q].value = xip[p]
+                    for p, q in enumerate(ind_xim):
+                        s.data[q].value = xim[p]
+
+        
+        if "galaxy_density_xi" in s.get_data_types():
+            for i in range(nbin_lens):
+                print(f"Computing theory density-density ({i},{i})")
 
                 # compute theory
-                print(tracers[f"source_{i}"], tracers[f"source_{j}"])
                 cl = pyccl.angular_cl(
-                    cosmo, tracers[f"source_{i}"], tracers[f"source_{j}"], ell
-                )
-                theta, *_ = s.get_theta_xi(
-                    "galaxy_shear_xi_plus", f"source_{i}", f"source_{j}"
-                )
-                xip = pyccl.correlation(cosmo, ell, cl, theta / 60, corr_type="L+")
-                xim = pyccl.correlation(cosmo, ell, cl, theta / 60, corr_type="L-")
-
-                # replace data values in the sacc object for the theory ones
-                ind_xip = s.indices(
-                    "galaxy_shear_xi_plus", (f"source_{i}", f"source_{j}")
-                )
-                ind_xim = s.indices(
-                    "galaxy_shear_xi_minus", (f"source_{i}", f"source_{j}")
-                )
-                for p, q in enumerate(ind_xip):
-                    s.data[q].value = xip[p]
-                for p, q in enumerate(ind_xim):
-                    s.data[q].value = xim[p]
-
-        for i in range(nbin_lens):
-            print(f"Computing theory density-density ({i},{i})")
-
-            # compute theory
-            cl = pyccl.angular_cl(
-                cosmo, tracers[f"lens_{i}"], tracers[f"lens_{i}"], ell
-            )
-            theta, *_ = s.get_theta_xi("galaxy_density_xi", f"lens_{i}", f"lens_{i}")
-            wtheta = pyccl.correlation(cosmo, ell, cl, theta / 60, corr_type="GG")
-
-            for j in range(i + 1):
-                print(f"Computing theory density-density ({i},{j})")
-
-                # compute theory
-                cl = pyccl.angular_cl(
-                    cosmo, tracers[f"lens_{i}"], tracers[f"lens_{j}"], ell
-                )
-                theta, *_ = s.get_theta_xi(
-                    "galaxy_density_xi", f"lens_{i}", f"lens_{j}"
+                    cosmo, tracers[f"lens_{i}"], tracers[f"lens_{i}"], ell
```

Comment:
It looks like this first set of cl aren't actually used, since the ones in the loop are instead?

Reply:
True! Good catch, I removed that.